Yoast/javascript
# Change KeywordInput and SynonymsInput's id prop to be required

## Summary

This PR can be summarized in the following changelog entry:

* [components] Changes `SynonymsInput`'s `id` prop to be required.
* [yoast-components] Changes `KeywordInput`'s `id` prop to be required.
* [non-userfacing search-metadata-previews] Removes unused `uniqueId` call.

## Relevant technical choices:

* _In hindsight_ `lodash`'s `uniqueId` is not handy to use in component Ids because snapshots and other tests need to get updated if the amount of times it is called changes.

## Test instructions

This PR can be tested by following these steps:

* Plugin test instructions are in [that PR](https://github.com/Yoast/wordpress-seo-premium/pull/2367).
* For this code one can check the component app if the KeywordInputs and SynonymInputs still function normally.
* If one wants to be _really_ strict also all the other components that use `uniqueId`:
	- `SettingsSnippetEditorFields` - as there the call got removed.
	- `SnippetEditorFields` uses it for the slug and label.
	- `HelpTextWrapper` uses it for the panel id.
	- `ReplacementVariableEditor` uses it for the editor and label.
	- Probably unrelated, but still uses it: the content analysis app's `SelectTestPaper`.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * [See the premium PR](https://github.com/Yoast/wordpress-seo-premium/pull/2367).

## UI changes

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2335
